### DISCUSSION TITLE
How to pass fixtures in parametrize to use it as list of params ?

### DISCUSSION THREAD

**Discussion by masudr4n4** [pytest-dev/pytest#8178] [Q&A]
```
def get_list():
    print('get_list function called')
    return [1,2,3,4]

@pytest.mark.parametrize("product",get_list())
def test_number(product):
    assert type(product) == int
```

this test case run successfully when i run this test separately but the problem is if i run other test from the same folder then this "get_list" function called every time too.I know this 'get_list' should be mark as fixtures but when i try to send the function name on parametrize then its gives me function objects is not iterable error.
how can i get rid of this? 
I want to call this function every time i run other test.
or how to treat this as fixtures to use it in parametrize because the list is dynamic in my real work.

```
@pytest.fixture()
def get_list():
    print('get_list function called')
    return [1,2,3,4]


@pytest.mark.usefixtures('get_list')
@pytest.mark.parametrize("product",get_list)
def test_number(product):
    assert type(product) == int
```
I tried like this too but its failed. The get_list function is not calling other test running but test_number test is not working ,its giving me function objects is not iterable error.

Thank in Advance.

**Comment by The-Compiler** [comment]
You could probably use a [parametrized fixture](https://docs.pytest.org/en/stable/fixture.html#fixture-parametrize) instead.

**Reply by masudr4n4** [reply]
@The-Compiler How can i do that? 
I am creating the params value dynamically,every time i run the test it will generate the list from api response.Where i will generate the list 
 to pass on parametrized fixture? 
Can you give me an short example?

**Reply by RonnyPfannschmidt** [reply]
https://docs.pytest.org/en/stable/parametrize.html?highlight=generate_tests#basic-pytest-generate-tests-example

**Reply by masudr4n4** [reply]
but for this i need to pass the paramerters value manuallly in terminal.

**Reply by RonnyPfannschmidt** [reply]
actual values of fixtures are not avaliale as source for parameter sets in pytest core, - i believe pytest-cases adds extra tools to support that until pytest-core does

**Comment by haroldmeerwaldt** [comment]
I got here because of the title, but I think my use case was slightly different. I had a fixture and I wanted to use that to parametrize a test. Something like:

```python
@pytest.fixture()
def my_list():
    print('get_list function called')
    return [1,2,3,4]

@pytest.mark.parametrize("product", my_list)
def test_number(product):
    assert type(product) == int
```
But then with my_list in conftest.py. This didn't work, also not variations with `request.usefixturevalue`.

In the end, I refactored a bit up, putting all the functionality of the fixture in a function and then using a parametrized fixture, something like:
```python
def get_list():
    print('get_list function called')
    return [1,2,3,4]

@pytest.fixture()
def my_list():
    return get_list()

@pytest.fixture(params=get_list())
def product(request):
    return request.param

def test_number(product):
    assert type(product) == int

def test_other_thing(my_list):
    assert len(my_list) == 4
```

**Comment by RonnyPfannschmidt** [comment]
Parametrize Happens before fixtures Are available
